From e5af875c1f092f33f6b8366b9fcd74fa18cdb125 Mon Sep 17 00:00:00 2001
From: "Kumar, Aman" <aman.kumar@intel.com>
Date: Thu, 10 Dec 2020 23:33:27 +0530
Subject: [PATCH 1/1] Fix for PSE UART Auto Flow Control

Added a small snippet of code from dw.c which has implementation about
changing the state of RTS and CTS.

Signed-off-by: Kumar, Aman <aman.kumar@intel.com>
---
 drivers/tty/serial/8250/8250_lpss.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/tty/serial/8250/8250_lpss.c b/drivers/tty/serial/8250/8250_lpss.c
index 47ad2e5bfc48..563b8a1d87be 100644
--- a/drivers/tty/serial/8250/8250_lpss.c
+++ b/drivers/tty/serial/8250/8250_lpss.c
@@ -177,7 +177,15 @@ static int byt_serial_setup(struct lpss8250 *lpss, struct uart_port *port)
 
 	return 0;
 }
+static void ehl_set_termios(struct uart_port *p, struct ktermios *termios,
+			    struct ktermios *old)
+{
+	p->status &= ~UPSTAT_AUTOCTS;
+	if (termios->c_cflag & CRTSCTS)
+		p->status |= UPSTAT_AUTOCTS;
 
+	serial8250_do_set_termios(p, termios, old);
+}
 static int ehl_serial_setup(struct lpss8250 *lpss, struct uart_port *port)
 {
 	struct dw_dma_slave *param = &lpss->dma_param;
@@ -188,6 +196,8 @@ static int ehl_serial_setup(struct lpss8250 *lpss, struct uart_port *port)
 	param->dma_dev = &pdev->dev;
 	lpss->dma_maxburst = 8;
 
+	port->set_termios = ehl_set_termios;
+
 	return 0;
 }
 
-- 
2.17.1

